### DISCUSSION TITLE
Get all state without AggregateId filter

### DISCUSSION THREAD

**Discussion by shashikantpawar21** [Eventuous/eventuous#120] [Q&A]
OrderAggregate --> OrderState, OrderId 

We got a requirement where we need restrict only 3 order to be created per day by the specific user.. 
To do this we **need to get all the orderstate without filtering by orderId in  order aggregate**
and check for the condition and then only allow the event to be applied.. 

But currently i can only get OrderState of particular orderId in OrderAggregate...Is there way to get all order state without OrderId filter ?

**Comment by JulianMay** [comment]
An aggregate is a transaction boundary, and your requirement crosses your current boundaries. This is one of the most typically asked questions regarding event sourcing (and distributed systems), not specific to Eventous, and you can find many answers online.
Your proposed solution (querying how many orders have been created by that user this day) can only be implemented using a projected read-model. 
Alternative solutions could be to change your aggregate boundaries (one aggregate per day with orders per customer, using customer-id and date as a composite key), or use a saga/process manager to implement the reservation-pattern. Again, these are all general solutions to general problems out there, not specific to Eventous.

**Reply by shashikantpawar21** [reply]
Thank you Julian with detail .. information .. we are going with projected read-model for now

**Reply by shashikantpawar21** [reply]
Julian ...Could you help us with a suggestion on below  query also .. It is causing us as a blocker in work
https://github.com/Eventuous/eventuous/discussions/119

**Reply by JulianMay** [reply]
No, I don't believe I can
